game/src/main/kotlin/gg/rsmod/game/sync/task/NpcSynchronizationTask.kt

```diff
@@ -49,10 +49,20 @@ class NpcSynchronizationTask(
     private fun getSegments(player: Player): List<SynchronizationSegment> {
         val segments = mutableListOf<SynchronizationSegment>()
 
-        val localNpcs = player.localNpcs
+        if (player.movedToInstance) {
+            player.localNpcs.clear()
+            segments.add(NpcCountSegment(0))
+            return segments
+        }
+
+        var localNpcs = player.localNpcs
         val iterator = localNpcs.iterator()
 
-        segments.add(NpcCountSegment(localNpcs.size))
+        if (player.movedFromInstance) {
+            segments.add(NpcCountSegment(0))
```

Comment:
I think this only works if there are no NPCs in the instanced area. Will need to do more testing here